lib/http/2/stream.rb

```diff
@@ -97,14 +101,33 @@ def receive(frame)
 
       case frame[:type]
       when :data
-        window_size = frame[:payload].bytesize
-        window_size += frame[:padding] || 0
-        @local_window -= window_size
+        # Emit DATA frame
+        delta_size = frame[:payload].bytesize
+        delta_size += frame[:padding] || 0
         emit(:data, frame[:payload]) unless frame[:ignore]
 
-        # Automatically send WINDOW_UPDATE,
-        # assuming that emit(:data) can now receive next data
-        window_update(window_size) if window_size > 0
+        @local_window -= delta_size
+        recv_window_size = @local_window_size - @local_window
```

Comment:
recv_window_size -> local_window_used, or some such? we're switching nomenclature for no reason.

Reply:
again, "inherited" naming from `nghttp`. `local_window_used` is clearer.